doc/user_guide/big_data.rst

```diff
@@ -0,0 +1,228 @@
+:py:class:`~._signals.lazy.LazySignal` and its derivatives. The main idea of
+these classes is to perform any operation (as the name suggests)
+`lazily <https://en.wikipedia.org/wiki/Lazy_evaluation>`_ (delaying the
+execution until the result is requested (e.g. saved, plotted)) and in a
+`blocked fashion <https://en.wikipedia.org/wiki/Block_matrix>`_. This is
+achieved by building a "history tree" (formally called a Directed Acyclic Graph
+(DAG)) of the computations, where the original data is at the root, and any
+further operations branch from it. Only when a certain branch result is
+requested, the way to the root is found and evaluated in the correct sequence
+on the correct blocks.
+
+The "magic" is performed by (for the sake of simplicity) storing the data not
+as ``numpy.ndarray``, but ``dask.array.Array`` (more information `here
+<https://dask.readthedocs.io/en/latest/>`_). ``dask`` offers a couple of
+advantages:
+
+* **Teoretically, arbitrary-sized data processing is possible**. By only
+  loading a couple of chunks at a time, theoretically any signal can be
+  processed, albeit slower. In practice, this may be limited: (i) some
+  operations may require certain chunking pattern, which may still saturate
+  memory; (ii) a couple of tens of chunks should fit into the computer memory
+  comfortably.
+* **Loading only the required data**. If a certain part (chunk) of the data is
+  not required for the final result, it will not be loaded at all, saving time
+  and resources.
+* **Able to extend to clusters**. ``dask.distributed`` (documentation `here
+  <https://distributed.readthedocs.io/en/latest/>`_) offers a straightforward
+  way to expand the effective memory for computations to that of a cluster,
+  which allows performing the operations significantly faster than on a single
+  machine.
+
+
+Creating Lazy Signals
+---------------------
+
+Loading lazily
+^^^^^^^^^^^^^^
+
+To load the data lazily, pass the keyword ``lazy=True``. Note that this
+deprecates and replaces all ``memmap`` and ``load_to_memory=False`` and similar
+options. As an example, loading a 34.9 GB ``.blo`` file on a regular laptop
+might look like:
+
+.. code-block:: python
+
+    >>> s = hs.load("shish26.02-6.blo", lazy=True)
+    >>> s
+    <LazySignal2D, title: , dimensions: (400, 333|512, 512)>
+    >>> s.data
+    dask.array<array-e..., shape=(333, 400, 512, 512), dtype=uint8, chunksize=(20, 12, 512, 512)>
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    uint8 34.9175808
+    >>> s.change_dtype("float") # To be able to perform decomposition, etc.
+    >>> print(s.data.dtype, s.data.nbytes / 1e9)
+    float64 279.3406464
+
+Only to load the data in the original unsigned integer format it would require
+around 35GB of memory. To store it in a floating-point format one would need
+almost 280GB of memory. Nevertheless, with the lazy processing both of these
+steps are near instantaneous and require very little resources.
```

Comment:
"Loading the dataset in the original unsigned integer format would require around 35GB of memory. To store it in a floating-point format one would need almost 280GB of memory. However, with lazy processing both of these steps are near-instantaneous and require very little computational resources."